Assets/Scripts/Player/PlayerJump.cs

```diff
@@ -0,0 +1,54 @@
+﻿using UnityEngine;
+
+[RequireComponent(typeof(CharacterController), typeof(PlayerGroundCheck))]
+public class PlayerJump : MonoBehaviour
+{
+    [SerializeField] private float m_jumpHeight = 5f;
+    [SerializeField] private float m_gravity = 10f;
+
+    private PlayerGroundCheck m_groundCheck;
+    private CharacterController m_controller;
+    private Vector3 m_velocity;
+
+    private void Start()
+    {
+        m_groundCheck = GetComponent<PlayerGroundCheck>();
```

Comment:
Don't initialize any code directly in monobehaviour methods, don't write any code directly in those methods at all. Create separate functions for it.